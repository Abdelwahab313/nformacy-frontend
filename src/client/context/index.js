import React, { createContext, useContext, useReducer } from 'react';
import {
  SET_ERROR_MESSAGE,
  UPDATE_CLIENTS,
  UPDATE_CURRENT_CLIENT,
  VERIFY_CLIENT,
  VERIFY_CURRENT_CLIENT,
} from './actionTypes';

const ClientContext = createContext();
const defaultValues = {
  clients: [],
  errorMessage: '',
};

const ClientProvider = ({ children, initialValue }) => {
  const { Provider } = ClientContext;
  initialValue = { ...defaultValues, ...initialValue };
  return (
    <Provider value={useReducer(clientReducer, initialValue)}>
      {children}
    </Provider>
  );
};

const useClientState = () => {
  const context = useContext(ClientContext);
  if (context === undefined) {
    throw new Error('useClientState must be used within a ClientProvider');
  }
  return context;
};

const clientReducer = (state, action) => {
  switch (action.type) {
    case UPDATE_CLIENTS:
      return {
        ...state,
        clients: action.payload,
      };
    case UPDATE_CURRENT_CLIENT:
      return {
        ...state,
        currentClient: action.payload,
      };
    case VERIFY_CURRENT_CLIENT:
      const currentClient = state.currentClient;
      currentClient.verified = true;
      return {
        ...state,
        currentClient: currentClient,
      };
    case VERIFY_CLIENT:
      const newClients = state.clients.map((client) => {
        if (client[0] === action.payload.uuid) {
          client[6] = true;
          return client;
        } else {
          return client;
        }
      });
      return {
        ...state,
        clients: newClients,
      };
    case SET_ERROR_MESSAGE:
      return {
        ...state,
        errorMessage: action.payload,
      };
  }
};

export { ClientProvider, useClientState };
