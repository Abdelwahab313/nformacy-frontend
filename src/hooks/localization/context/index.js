import React, { createContext, useContext, useReducer } from 'react';

const LocaleContext = createContext();
const DEFAULT_LOCALE = 'en';
const LOCALES = {
  en: 'en',
  ar: 'ar',
};
export const CHANGE_LOCALE = 'CHANGE_LOCALE';

export const useLocaleContext = () => {
  const context = useContext(LocaleContext);
  if (context === undefined)
    throw new Error('useLocale should only be used within locale provider');
  return context;
};

export const LocaleProvider = ({
  children,
  initialLocale = DEFAULT_LOCALE,
}) => {
  const { Provider } = LocaleContext;
  return (
    <Provider value={useReducer(localeReducer, initialLocale)}>{children}</Provider>
  );
};

const localeReducer = (state, action) => {
  if (action.type === CHANGE_LOCALE) {
    return action.payload.locale;
  }
  return LOCALES.en;
};
