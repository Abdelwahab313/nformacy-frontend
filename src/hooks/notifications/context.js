import React, { useContext, useReducer } from 'react';
import { ActionCableProvider } from '../useActionCable';
import { CHANNEL_URL } from '../../settings';
import { produce, current } from 'immer';
import { toast as showToast } from 'react-toastify';

const defaultStates = {
  notifications: [],
  menuOpened: null,
  unread: false,
  unreadCount: 0,
};

const NotificationsContext = React.createContext();

export const notificationActions = {
  notificationReceived: 'notificationReceived',
  menuToggled: 'menuToggled',
  menuClosed: 'menuClosed',
};

const MAX_NOTIFICATIONS = 10;
export const useNotificationsContext = () => {
  const context = useContext(NotificationsContext);
  if (context === undefined) {
    throw new Error(
      'useNotificationsContext must be used within a NotificationsProvider',
    );
  }
  return context;
};
const initContext = (initialValue) => {
  return initialValue || defaultStates;
};

export const NotificationsProvider = ({
  children,
  initialNotifications = [],
  unreadCount = 0,
}) => {
  const { Provider } = NotificationsContext;
  const initialState = {
    ...defaultStates,
    notifications: initialNotifications,
    unread: unreadCount > 0,
    unreadCount,
  };
  return (
    <Provider
      value={useReducer(NotificationsReducer, initialState, initContext)}>
      <ActionCableProvider url={CHANNEL_URL}>{children}</ActionCableProvider>
    </Provider>
  );
};

function Notification(notification) {
  return {
    notificationId: notification.notification_id,
    targetId: notification.target_id,
    messageKey: notification.message_key,
    createdAt: notification.created_at,
    type: notification.type,
    readAt: null,
  };
}

const updateUserInLocalStorage = (currentUser, updatedFields) => {
  localStorage.setItem(
    'user',
    JSON.stringify({ ...currentUser, ...updatedFields }),
  );
};

const NotificationsReducer = (state, action) => {
  switch (action.type) {
    case notificationActions.notificationReceived:
      const currentUser = action.payload.currentUser;
      const receivedNotification = Notification(action.payload.notification);
      showToast(receivedNotification.messageKey, {
        toastId: receivedNotification.notificationId,
      });
      return produce(state, (draftState) => {
        if (draftState.notifications.length === MAX_NOTIFICATIONS) {
          draftState.notifications.pop();
        }
        draftState.notifications.unshift(receivedNotification);
        draftState.unread = true;
        draftState.unreadCount = state.unreadCount + 1;
        updateUserInLocalStorage(currentUser, {
          unreadNotifications: draftState.unreadCount,
          notifications: current(draftState.notifications),
        });
      });
    case notificationActions.menuToggled:
      return produce(state, (draftState) => {
        if (state.menuOpened?.contains(action.payload.target)) {
          draftState.menuOpened = null;
        } else {
          draftState.menuOpened = action.payload.currentTarget;
          draftState.unread = false;
        }
      });
    case notificationActions.menuClosed:
      return produce(state, (draftState) => {
        draftState.menuOpened = null;
      });
    default:
      throw Error(
        `Action: ${action.type} is not defined in notifications reducer`,
      );
  }
};
