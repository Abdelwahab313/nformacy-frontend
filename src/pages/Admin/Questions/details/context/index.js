import React, { createContext, useContext, useReducer } from 'react';

const defaultValues = {
  questionDetails: '',
};

const QuestionContext = createContext();

const QuestionProvider = ({ children, initialValue = {} }) => {
  initialValue = { ...defaultValues, ...initialValue };
  const { Provider } = QuestionContext;
  return (
    <Provider value={useReducer(QuestionReducer, initialValue)}>
      {children}
    </Provider>
  );
};

const useQuestionContext = () => {
  const context = useContext(QuestionContext);
  if (context === undefined) {
    throw new Error('useQuestionContext must be used within a AuthProvider');
  }
  return context;
};

export const QuestionActionTypes = {
  UPDATE_QUESTION: 'UPDATE_QUESTION',
};

const QuestionReducer = (state, action) => {
  switch (action.type) {
    case QuestionActionTypes.UPDATE_QUESTION:
      return { ...state, questionDetails: action.payload };

    default:
      throw new Error(`Unhandled action type: ${action.type}`);
  }
};

export { useQuestionContext as useAuth, QuestionProvider as AuthProvider };
