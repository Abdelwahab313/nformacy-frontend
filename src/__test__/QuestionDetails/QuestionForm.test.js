import React from 'react';
import { render } from '@testing-library/react';
import QuestionForm from 'pages/Admin/Questions/QuestionDetails/subComponents/QuestionForm';
import { AuthProvider } from 'pages/auth/context/auth';
import moment from 'moment';
import { QuestionProvider } from 'pages/Admin/Questions/QuestionDetails/context';

const testQuestion = {
  id: 2,
  referenceNumber: 2000131,
  title: 'Test mocked question',
  content:
    "<p>People are different. People choose different criteria. But if there is a better way among many alternatives, I want to encourage that way by making it comfortable. So that's what I've tried to do.</p>",
  field: [{ value: 'marketingAndPR', label: 'Marketing and PR' }],
  subfield: [{ value: 'marketResearch', label: 'Market Research' }],
  industry: { value: 'consulting', label: 'Consulting' },
  mediaId: null,
  assignmentType: 'call',
  currentActionTime: moment().add(5, 'days'),
  createdAt: '2020-09-14T09:25:55.704Z',
  hoursToCloseAnswers: 100,
  hoursToReviewAndEdit: 24,
  assignedAdviserId: 2,
  state: 'pending_adviser_acceptance',
};

describe('Question Form', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <AuthProvider>
        <QuestionProvider>
          <QuestionForm questionDetails={testQuestion} />
        </QuestionProvider>
      </AuthProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
