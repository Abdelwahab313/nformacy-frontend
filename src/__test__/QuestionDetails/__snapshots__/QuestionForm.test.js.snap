// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Form should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="cardBody"
  >
    <div
      class="MuiGrid-root grid MuiGrid-container"
    >
      <div
        class="MuiGrid-root grid MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-2"
      >
        <div
          class="MuiFormControl-root undefined formControl MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root labelRootundefined MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
            for="reference-id"
          >
            Reference ID
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline underline Mui-disabled Mui-disabled disabled MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled disabled"
              disabled=""
              id="reference-id"
              name="referenceNumber"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root grid MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-7"
      >
        <div
          class="MuiFormControl-root undefined formControl MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root labelRootundefined MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
            for="title"
          >
            Title
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline underline MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              id="title"
              name="title"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root grid MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-3"
      >
        <div
          class="MuiFormControl-root undefined formControl MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root labelRootundefined MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
            data-shrink="true"
            for="post-date"
          >
            Post Date
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline underline Mui-disabled Mui-disabled disabled MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled disabled"
              disabled=""
              id="post-date"
              name="createdAt"
              type="text"
              value="NaN years ago"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root grid inputsRow MuiGrid-container"
    >
      <div
        class="MuiGrid-root grid MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-3"
      >
        <div
          aria-expanded="false"
          class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
          name="majorFieldsOfExperience"
          role="combobox"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
              for="majorFieldsOfExperienceSelect"
              id="majorFieldsOfExperienceSelect-label"
            >
              majorFields
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot makeStyles-inputRoot-1 MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
            >
              <input
                aria-autocomplete="list"
                aria-invalid="false"
                autocapitalize="none"
                autocomplete="off"
                class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                id="majorFieldsOfExperienceSelect"
                spellcheck="false"
                type="text"
                value=""
              />
              <div
                class="MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Clear"
                  class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
                  tabindex="-1"
                  title="Clear"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-label="Open"
                  class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                  tabindex="-1"
                  title="Open"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-4"
                >
                  <span>
                    majorFields
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root grid MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-3"
      >
        <div
          aria-expanded="false"
          class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
          name="fields"
          role="combobox"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
              for="specificFieldsOfExperienceSelect"
              id="specificFieldsOfExperienceSelect-label"
            >
              specificField
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot makeStyles-inputRoot-1 MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
            >
              <input
                aria-autocomplete="list"
                aria-invalid="false"
                autocapitalize="none"
                autocomplete="off"
                class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                id="specificFieldsOfExperienceSelect"
                spellcheck="false"
                type="text"
                value=""
              />
              <div
                class="MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Clear"
                  class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
                  tabindex="-1"
                  title="Clear"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-label="Open"
                  class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                  tabindex="-1"
                  title="Open"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-4"
                >
                  <span>
                    specificField
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root grid MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-3"
      >
        <div
          aria-expanded="false"
          class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
          name="industry"
          role="combobox"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
              for="industry"
              id="industry-label"
            >
              Industry
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
            >
              <input
                aria-autocomplete="list"
                aria-invalid="false"
                autocapitalize="none"
                autocomplete="off"
                class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                id="industry"
                spellcheck="false"
                type="text"
                value=""
              />
              <div
                class="MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Clear"
                  class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
                  tabindex="-1"
                  title="Clear"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-label="Open"
                  class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                  tabindex="-1"
                  title="Open"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-4"
                >
                  <span>
                    Industry
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root grid MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-3"
      >
        <div
          aria-expanded="false"
          class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
          name="QuestionLanguage"
          role="combobox"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
              for="questionLanguage"
              id="questionLanguage-label"
            >
              Question Language
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
            >
              <input
                aria-autocomplete="list"
                aria-invalid="false"
                autocapitalize="none"
                autocomplete="off"
                class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                id="questionLanguage"
                spellcheck="false"
                type="text"
                value=""
              />
              <div
                class="MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Clear"
                  class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
                  tabindex="-1"
                  title="Clear"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-label="Open"
                  class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                  tabindex="-1"
                  title="Open"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-4"
                >
                  <span>
                    Question Language
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root grid inputsRow MuiGrid-container"
    >
      <div
        class="MuiGrid-root grid MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-3"
      >
        <div
          aria-expanded="false"
          class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
          name="AssignmentType"
          role="combobox"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
              for="assignmentType"
              id="assignmentType-label"
            >
              Type of Assignment
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
            >
              <input
                aria-autocomplete="list"
                aria-invalid="false"
                autocapitalize="none"
                autocomplete="off"
                class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                id="assignmentType"
                spellcheck="false"
                type="text"
                value=""
              />
              <div
                class="MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Clear"
                  class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
                  tabindex="-1"
                  title="Clear"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-label="Open"
                  class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                  tabindex="-1"
                  title="Open"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-4"
                >
                  <span>
                    Type of Assignment
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root grid countDownContainer MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-3"
      >
        <div
          class="MuiFormControl-root undefined formControl MuiFormControl-fullWidth"
          style="margin: 0px;"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root labelRootundefined MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
            for="hoursToCloseAnswers"
          >
            Time for Freelancers to Answer (In Hours)
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline underline MuiInputBase-formControl MuiInput-formControl"
            style="padding-bottom: 12px;"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              id="hoursToCloseAnswers"
              name="hoursToCloseAnswers"
              type="number"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root grid countDownContainer MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-3"
      >
        <div
          class="MuiFormControl-root undefined formControl MuiFormControl-fullWidth"
          style="margin: 0px;"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root labelRootundefined MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
            for="hoursToReviewAndEdit"
          >
            Time for Adviser to Review (In Hours)
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline underline MuiInputBase-formControl MuiInput-formControl"
            style="padding-bottom: 12px;"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              id="hoursToReviewAndEdit"
              name="hoursToReviewAndEdit"
              type="number"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root grid MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-3"
      >
        <div
          aria-expanded="false"
          class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
          name="AssignAdviser"
          role="combobox"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
              for="assignAdviser"
              id="assignAdviser-label"
            >
              Assign Adviser
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
            >
              <input
                aria-autocomplete="list"
                aria-invalid="false"
                autocapitalize="none"
                autocomplete="off"
                class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                id="assignAdviser"
                spellcheck="false"
                type="text"
                value=""
              />
              <div
                class="MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Clear"
                  class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator MuiAutocomplete-clearIndicatorDirty"
                  tabindex="-1"
                  title="Clear"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-label="Open"
                  class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                  tabindex="-1"
                  title="Open"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-4"
                >
                  <span>
                    Assign Adviser
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root grid inputsRow MuiGrid-container"
    >
      <div
        class="MuiGrid-root grid MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root contentTitle MuiInputLabel-animated"
        >
          Question Content
        </label>
        <div
          class="MuiGrid-root questionContainer MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <textarea
              id="richContent"
              style="visibility: hidden;"
            />
          </div>
          <div
            class="MuiGrid-root undefined attachmentContainer MuiGrid-item MuiGrid-grid-xs-6"
          />
          <div
            class="MuiGrid-root answerButtonContainer MuiGrid-item MuiGrid-grid-xs-6"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained answerButtons"
              id="applyChangesButton"
              style="background-color: rgb(18, 87, 115); color: rgb(255, 255, 255);"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Apply Changes
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
